src/verify_cert.rs

```diff
@@ -431,20 +534,20 @@ const EKU_SERVER_AUTH: KeyPurposeId = KeyPurposeId::new(&oid!(1, 3, 6, 1, 5, 5,
 // id-kp-clientAuth   OBJECT IDENTIFIER ::= { id-kp 2 }
 const EKU_CLIENT_AUTH: KeyPurposeId = KeyPurposeId::new(&oid!(1, 3, 6, 1, 5, 5, 7, 3, 2));
 
-fn loop_while_non_fatal_error<V>(
+fn loop_while_non_fatal_error<'a, V: 'a>(
     default_error: Error,
     values: V,
-    mut f: impl FnMut(V::Item) -> Result<(), ControlFlow<Error, Error>>,
-) -> Result<(), ControlFlow<Error, Error>>
+    mut f: impl FnMut(V::Item) -> Result<&'a TrustAnchor<'a>, ControlFlow<Error, Error>>,
+) -> Result<&'a TrustAnchor<'a>, ControlFlow<Error, Error>>
 where
     V: IntoIterator,
 {
     let mut error = default_error;
     for v in values {
         match f(v) {
-            Ok(()) => return Ok(()),
+            Ok(item) => return Ok(item),
```

Comment:
nit: `item` is not very descriptive.
```suggestion
            Ok(trust_anchor) => return Ok(trust_anchor),
```